src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -280,8 +285,134 @@ py::array array_from_tensor(ov::Tensor&& t, bool is_shared) {
+    // Get actual dtype from OpenVINO type:
+    auto ov_type = c.get_element_type();
+    auto dtype = Common::type_helpers::get_dtype(ov_type);
+    // Return the array as a view:
+    if (is_shared) {
+        if (ov_type.bitwidth() < Common::values::min_bitwidth) {
+            return py::array(dtype, c.get_byte_size(), c.get_data_ptr(), py::cast(c));
+        }
+        return py::array(dtype, c.get_shape(), constant_helpers::_get_strides(c), c.get_data_ptr(), py::cast(c));
+    }
+    // Return the array as a copy:
+    if (ov_type.bitwidth() < Common::values::min_bitwidth) {
+        return py::array(dtype, c.get_byte_size(), c.get_data_ptr());
+    }
+    return py::array(dtype, c.get_shape(), constant_helpers::_get_strides(c), c.get_data_ptr());
+}
+
 };  // namespace array_helpers
 
+namespace tensor_helpers {
+void fill_tensor(ov::Tensor& tensor, py::array& array) {
+    // If ndim of py::array is 0, array is a numpy scalar. That results in size to be equal to 0.
+    std::memcpy(tensor.data(),
+                array.ndim() == 0 ? array.data() : array.data(0),
+                array.ndim() == 0 ? array.itemsize() : array.nbytes());
+}
+
+void fill_tensor(ov::Tensor& tensor, py::array& array, ov::element::Type& dst_dtype) {
+    if (array.dtype().is(py::dtype("float64"))) {
+        fill_tensor_t<double>(tensor, array, dst_dtype);
+    } else if (array.dtype().is(py::dtype("float32"))) {
+        fill_tensor_t<float>(tensor, array, dst_dtype);
+    } else if (array.dtype().is(py::dtype("float16"))) {
+        fill_tensor_t<ov::float16>(tensor, array, dst_dtype);
+    } else if (array.dtype().is(py::dtype("int64"))) {
+        fill_tensor_t<int64_t>(tensor, array, dst_dtype);
+    } else if (array.dtype().is(py::dtype("int32"))) {
+        fill_tensor_t<int32_t>(tensor, array, dst_dtype);
+    } else if (array.dtype().is(py::dtype("int16"))) {
+        fill_tensor_t<int16_t>(tensor, array, dst_dtype);
+    } else if (array.dtype().is(py::dtype("int8"))) {
+        fill_tensor_t<int8_t>(tensor, array, dst_dtype);
+    } else if (array.dtype().is(py::dtype("uint64"))) {
+        fill_tensor_t<uint64_t>(tensor, array, dst_dtype);
+    } else if (array.dtype().is(py::dtype("uint32"))) {
+        fill_tensor_t<uint32_t>(tensor, array, dst_dtype);
+    } else if (array.dtype().is(py::dtype("uint16"))) {
+        fill_tensor_t<uint16_t>(tensor, array, dst_dtype);
+    } else if (array.dtype().is(py::dtype("uint8"))) {
+        fill_tensor_t<uint8_t>(tensor, array, dst_dtype);
+    } else if (array.dtype().is(py::dtype("bool"))) {
+        fill_tensor_t<bool>(tensor, array, dst_dtype);
+    } else {
+        OPENVINO_THROW("Unsupported dtype passed!");
+    }
+}
+
+};  // namespace tensor_helpers
+
+namespace constant_helpers {
+std::vector<size_t> _get_strides(const ov::op::v0::Constant& self) {
+    auto element_type = self.get_element_type();
+    auto shape = self.get_shape();
```

Comment:
```suggestion
    const auto& element_type = self.get_element_type();
    const auto& shape = self.get_shape();
```

the strides could be calculate like:
```
auto strides = ov::row_major_strides(s)
// then mul each stride in strides by element_type.size() 
```